python/chemiscope/input.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+"""
+Generate JSON input files for the default chemiscope visualizer.
+"""
+import warnings
+import json
+import gzip
+
+import numpy as np
+
+from .adapters import frames_to_json, atom_properties, structure_properties
+
+
+def _typetransform(data):
+    assert isinstance(data, list) and len(data) > 0
+    if isinstance(data[0], str):
+        return list(map(str, data))
+    elif isinstance(data[0], bytes):
+        return list(map(lambda u: u.decode("utf8"), data))
+    else:
+        try:
+            return [float(value) for value in data]
+        except ValueError:
+            raise Exception("unsupported type in property")
+
+
+def _linearize(name, property):
+    """
+    Transform 2D arrays in multiple 1D arrays, converting types to fit json as
+    needed.
+    """
+    data = {}
+    if isinstance(property["values"], list):
+        data[name] = {
+            "target": property["target"],
+            "values": _typetransform(property["values"]),
+        }
+    elif isinstance(property["values"], np.ndarray):
+        if len(property["values"].shape) == 1:
+            data[name] = {
+                "target": property["target"],
+                "values": _typetransform(list(property["values"])),
+            }
+        elif len(property["values"].shape) == 2:
+            if property["values"].shape[1] == 1:
+                data[name] = {
+                    "target": property["target"],
+                    "values": _typetransform(list(property["values"])),
+                }
+            else:
```

Comment:
Will this work for an array of lists?

Reply:
no, but you have to try pretty hard to create an array of list with numpy, manually setting `dtype=object`. I'll have to check, but I think this will fails with an assert later.

Reply:
Yep, this is caught by a check later.